lab-doug/test/receipt-router-test.js

```diff
@@ -0,0 +1,179 @@
+'use strict';
+//use test db for this module.  Make sure this line is before loading of server and the 'port' const
+process.env.MONGO_URI = 'mongodb://localhost/test';
+const expect = require('chai').expect;
+const request = require('superagent-use');
+const supPromise = require('superagent-promise-plugin');
+const port = process.env.PORT || 3000;
+const baseUrl = `http://localhost:${port}`;
+const server = require('../server');
+const receiptCrud = require('../lib/receipt-crud');
+const repairCrud = require('../lib/repair-crud');
+
+
+request.use(supPromise);
+
+describe('Testing RECEIPT router', function(){
+  before((done) => {
+    if(!server.isRunning){
+      server.listen(port, () => {
+        server.isRunning = true;
+        console.log('server is running on port: ', port);
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  after((done) => {
+    if(server.isRunning){
+      server.close(() => {
+        console.log('server has been shutdown');
+        done();
+      });
```

Comment:
I would also include something to tell mongo to drop the database in your after.